feature: creation of mcaf-budgets module (#1)
schubergphilis/terraform-aws-mcaf-budget: README.md

# terraform-aws-mcaf-budget

Terraform module to create and manage an AWS Budget.

> [!IMPORTANT]

> > There is a known limitation within AWS Budgets. AWS Budgets is not in complete parity with Cost Explorer. Currently, Budgets does not support charge types related to Savings Plans or reservation applied usage. This means that the spend you see in AWS Budgets can be different then you see in AWS Cost Explorer or your AWS bill.

Reviewer: ```suggestion
> Please note a limitation in AWS Budgets in comparison to Cost Explorer: AWS Budgets does not currently fully align with Cost Explorer, lacking support for charge types related to Savings Plans or reservation-applied usage. Consequently, the expenditure displayed in AWS Budgets may differ from what you observe in AWS Cost Explorer or your AWS bill.
```

> As a temporary work-around, you needs to select the following charge types in Cost Explorer after clicking on "View in Cost Explorer" in your AWS Budget:

Reviewer: ```suggestion
As a temporary workaround, you should choose the following charge types in Cost Explorer after clicking "View in Cost Explorer" in your AWS Budget:
```

- Usage (should already be selected)
- Reservation applied usage
- Savings Plan Covered Usage
- Savings Plan Negation
- Savings Plan Recurring Fee
- Savings Plan Upfront Fee

Now the Budgets/Cost Explorer values be the same. AWS expects Cloud Explorer parity to be complete by EOY 2024.

> [!TIP]

> > We do not pin modules to versions in our examples. We highly recommend that in your code you pin the version to the exact version you are using so that your infrastructure remains stable.

Reviewer: Can we remove this? This is a given and we shouldn't need to be explicitly mention it. 

If you really want to mention this, let's move it a readme in either `examples` or `examples/basic` to make the project readme less cluttered 👍

Author: Its like this in all repo's, I think its not very clutterering and mentioning this is quite common
